### ISSUE TITLE
Support `%p` expansion for `pylint` executable

### ISSUE THREAD

**Issue by posita** [AtomLinter/linter-pylint#62]
Even with #24, it might be nice to be able to do something like this:

![screen shot 2015-09-01 at 21 07 25](https://cloud.githubusercontent.com/assets/222581/9622858/9174a3ea-50ed-11e5-929e-f1574e700803.png)

But right now, despite `[PROJECT_DIR]/.venv/bin/pylint` existing, one still gets this error:

![screen shot 2015-09-01 at 21 03 23](https://cloud.githubusercontent.com/assets/222581/9622863/afefc50c-50ed-11e5-8030-d6f9d81b89b3.png)

**Comment by posita** [comment]
It might also be nice to use the specified executable if it exists, but degrade to one found in `${PATH}` if it doesn't. That would enable one to override `pylint` on a per project basis, but avoid errors if `pylint` wasn't present inside a project.

Alternately, maybe enable something like this?

`[firstpathtotry]:[secondpathtotry]:...:[nthpathtotry]`

Which would result in an attempt to find `pylint` in the following paths (in order):

1. `firstpathtotry`
1. `firstpathtotry/pylint`
1. `secondpathtotry`
1. `secondpathtotry/pylint`
1. ...
1. `nthpathtotry`
1. `nthpathtotry/pylint`

When combined with `%p` expansion, this would allow something like:

`%p/.venv/bin:/usr/local/bin:/usr/bin`

If environment variable expansion was also enabled, one could do:

`${PYLINT}:%p/.venv/bin/pylint3:%p/.venv/bin/pylint2:%p/.venv/bin:${PATH}`

**Comment by steelbrain** [comment]
cc @SpainTrain

**Comment by SpainTrain** [comment]
>  it might be nice to be able to do something like this

Definitely agree.

> It might also be nice to use the specified executable if it exists, but degrade to one found in ${PATH} if it doesn't.

Also agree with this.  I prefer this approach to the alternative of reimplementing the priority logic of finding executables in `$PATH`. Plus, I suspect that one would only need to have one executable path that uses the project directory, right?

If it's reasonable, I think we will plan for `%p` interpolation in executable with a backup to try `pylint`.

**Comment by posita** [comment]
> If it's reasonable, I think we will plan for `%p` interpolation in executable with a backup to try `pylint` [via `PATH`].

I think this handles all the important use cases. Even if the global `pylint` executable is named something weird on a particular system (e.g., `pylint-2.7`) a user can always do something like:

```sh
% mkdir -p "${HOME}/bin"
% ln -s "$( which pylint-2.7 )" "${HOME}/bin/pylint"
...
% export PATH="${HOME}/bin:${PATH:+:}${PATH}"
```

That work-around doesn't require `root` access. I think that covers all the bases. :+1:

**Comment by posita** [comment]
[sadovnychyi/autocomplete-python](/sadovnychyi/autocomplete-python) has a solution (for python binaries) that seems to work well. It expands `$PROJECT` and allows multiple paths split by `;` (e.g., `$PROJECT/.venv/bin;python2.7`. You might want to check out [source code](/sadovnychyi/autocomplete-python/blob/740238ee43222dd25a6ef7d3de5aa7275a7636f4/lib/provider.coffee#L92). It has [limitations](/sadovnychyi/autocomplete-python/issues/66), however. :wink: 

It might be nice to support other variable expansion/notation as well (e.g., `${HOME}`). Perhaps some collaboration is appropriate?

**Comment by SpainTrain** [comment]
This should now be closed by #91
